cds/modules/deposit/api.py

```diff
@@ -573,6 +575,33 @@ def _current_tasks_status(self):
         return get_tasks_status_by_task(
             get_deposit_events(self['_deposit']['id']))
 
+    @contextmanager
+    def _process_files(self, record_id, data):
+        """Snapshot bucket and add files in record during first publishing."""
+        if self.files:
+            from invenio_db import db
+            assert not self.files.bucket.locked
+            self.files.bucket.locked = True
+            snapshot = self.files.bucket.snapshot(lock=True)
+            # get all tags for objects in bucket
+            all_tags = [tag for obj in snapshot.objects for tag in obj.tags]
+            # find version_id of object with is_master tag
+            master_tag = [tag for tag in all_tags
+                          if tag.key == 'is_master' and tag.value == 'true']
+            master_version = next(iter(master_tag), None).version_id
+            # change master of slave videos to new master object
+            slave_tags = [tag for tag in all_tags if tag.key == 'master']
+            for tag in slave_tags:
+                tag.value = str(master_version)
```

Comment:
Like this you only contemplate one so called "master file" per record, but in fact we can have several, i.e. subtitles or transcripts

I would take profit of what `self.files.dumps` gives me and verify the `uuid` for every file in the first level